src/workload.rs

```diff
@@ -0,0 +1,182 @@
+use crate::config::Config;
+use crate::{
+    failpoint::{disable_failpoint, enable_failpoint},
+    table::Table,
+    Effectiveness, AVAILABLE_INJECTIONS,
+};
+use async_trait::async_trait;
+use mysql::prelude::*;
+use std::collections::HashMap;
+
+// a helper function that collects the results of an execution of a workload
+fn collect_result(
+    res: Result<(), mysql::Error>,
+    results: &mut HashMap<(Table, String, String), Effectiveness>,
+    table: &Table,
+    injection: &str,
+    conn: &mut mysql::PooledConn,
+) {
+    let e = match res {
+        Ok(_) => match conn.query_drop(format!("admin check table {}", table.name)) {
+            Ok(_) => Effectiveness::Consistent,
+            Err(_) => Effectiveness::NoError,
+        },
+        Err(x) if x.to_string().contains("inconsist") => Effectiveness::Inconsistent,
+        Err(_) => Effectiveness::OtherError,
+    };
+    results.insert(
+        (table.clone(), "workload_a".into(), injection.to_string()),
+        e,
+    );
+}
+
+// a helper function
+fn enable_featuers(
+    conn: &mut mysql::PooledConn,
+    config: &Config,
+) -> Result<(), Box<dyn std::error::Error>> {
+    conn.query_drop(format!(
+        "set @@tidb_enable_mutation_checker = {}",
+        config.mutation_checker
+    ))?;
+    conn.query_drop(format!(
+        "set @@tidb_txn_assertion_level = {}",
```

Comment:
```suggestion
        "set @@tidb_txn_assertion_level = \"{}\"",
```
I guess here it needs to be quoted since it's a string?

Reply:
I tried, and it works without quotes😂.
```sql
mysql> select @@tidb_txn_assertion_level;
+----------------------------+
| @@tidb_txn_assertion_level |
+----------------------------+
| FAST                       |
+----------------------------+
1 row in set (0.00 sec)

mysql> set @@tidb_txn_assertion_level = OFF;
Query OK, 0 rows affected (0.00 sec)

mysql> select @@tidb_txn_assertion_level;
+----------------------------+
| @@tidb_txn_assertion_level |
+----------------------------+
| OFF                        |
+----------------------------+
1 row in set (0.00 sec)
```